ui/admin-portal/src/app/model/chat.spec.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2024 Talent Beyond Boundaries.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {ChatPost, CreateChatRequest, JobChat, JobChatType, JobChatUserInfo, Post} from "./chat";
+import {User} from "./user";
+
+fdescribe('Job Chat and Post interface', () => {
+  it('should create a Post instance', () => {
+    const post: Post = {
+      id: 1,
+      content: 'Sample content',
+    };
+
+    expect(post).toBeTruthy();
+    expect(post.id).toBe(1);
+    expect(post.content).toBe('Sample content');
+  });
```

Comment:
I don't understand what logic that this is testing.

const post: Post = {
      id: 1,
      content: 'Sample content',
};

Doesn't run any of our logic, does it. It just populates an object which could be any object - and then checks that the fields that you just populated are in fact populated. There is no TC logic being tested here, is there?

Reply:
@camerojo 

I understand your concern regarding the current test. Models, by their nature, are simple data structures and don't contain much logic. Therefore, testing them typically involves verifying their properties.

The main logic involving models is usually tested in the components or services that use them. The current test ensures that the model is correctly instantiated with the expected properties, which is a necessary step before these models are used in more complex logic within components.

Reply:
Model tests checks that for example the JobChat object is correctly instantiated with the expected properties. If we change the JobChat model, for example, by renaming a property, changing its type, adding a new property, or removing an existing one, this test will fail. This ensures that any changes to the model structure are caught and reviewed.

Reply:
However, I agree that we should focus on testing the actual logic in the components (done) or services where these models are utilized. The current test is a foundational check, and the more complex behaviors should be tested in the context where they are applied.

Reply:
If you have specific scenarios or functionalities in mind that need thorough testing, please let me know, and I will ensure those are covered in our test suite.

Reply:
I am still not sure that we need such simple tests. For example, if someone renames a property or removes an existing one, that will cause a Typescript syntax error and the code won't compile.

If a new property is added, your test won't fail.

Reply:
Happy to approve it though - just a thought.

Reply:
I completely agree with you, Since these models are simple models with no methods or logic, unit tests focusing on its properties won't add much value. However, I can test how this model is used within the larger application context, such as within services or components that manipulate or validate Model instances.

Currently, about 10 models remained, which I plan to implement in a separate ticket. Before proceeding, I would like to get your input: Should I go ahead with writing these tests, or would it be better to stop?

Reply:
However, I want to create .spec files to maintain consistency in testing structure, I can create basic tests to ensure the model's properties are correctly instantiated.
What do you think?